spec/rspec/core/world_spec.rb

```diff
@@ -136,6 +136,19 @@ module RSpec::Core
       context "when --only-failures is passed" do
         before { configuration.force(:only_failures => true) }
 
+        context "and all examples are filtered out" do
+          before do
+            configuration.filter_run_including :foo => 'bar'
+          end
+
+          it 'will ignore run_all_when_everything_filtered' do
+            allow(configuration).to receive(:run_all_when_everything_filtered?) { true }
```

Comment:
There's no need to reset the config after, because `configuration` is defined by a `let` and is a new instance in each example -- so there's no leakage automatically.

> I cribbed this from the existing tests

Bad patterns in existing tests shouldn't be propagated to new tests.  Stubbing is more brittle than just setting `configuration.run_all_when_everything_filtered` since I believe that there are two ways to access the value -- `config.run_all_when_everything_filtered` and `config.run_all_when_everything_filtered?`.  Plus we're not using it to design the config API here.